jobs/dap-collector-ppa-prod/dap_collector_ppa_prod/main.py

```diff
@@ -208,14 +222,55 @@ async def collect_many(
 
 
 async def collect_task(task, auth_token, hpke_private_key, date):
-    """Collects data for the given task and the given day."""
-    start = datetime.datetime.fromisoformat(date)
-    start = start.replace(tzinfo=datetime.timezone.utc)
-    end = start + datetime.timedelta(minutes=JOB_INTERVAL)
+    """Collects data for the given task through to the given day.
+        For tasks with time precision smaller than a day, will collect data for aggregations from the day prior to date.
+        For tasks with time precision a day or multiple of day, will collect data for the aggregation that ends on date.
+            If date does not align with the end of an aggregation, it will not collect anything.
+    """
+    until = datetime.datetime.fromisoformat(date)
+    until = until.replace(tzinfo=datetime.timezone.utc)
+    time_precision_minutes = task["time_precision_minutes"]
 
-    results = await collect_many(
-        task, start, end, INTERVAL_LENGTH, hpke_private_key, auth_token
-    )
+    until_seconds = int(until.timestamp())
+
+    if until.hour != 0 or until.minute != 0:
```

Comment:
This logic block seems accurate to me, though I think it might read better if the error scenarios were moved into their own function and if an error message is returned, then this function can exit and return the error result, otherwise it can check which valid scenario it needs to collect.

Similar to:
err = checkTimePrecision(time_precision_minutes)
if err is not None:
    return build_error_result(task["task_id"], until, task["metric_type"], err);

# Check valid collection scenario

Reply:
this also made it occur to me having non-zero seconds would also be a problem